permits/forms.py

```diff
@@ -619,6 +611,34 @@ def __init__(self, *args, **kwargs):
             ]
             self.fields["status"].choices = tuple(filter2)
 
+            for field in services.get_permit_request_amend_custom_properties(
+                self.instance
+            ):
+                field_name = f"{models.AMEND_CUSTOM_FIELDS_PREFIX}{field.id}"
+                self.fields[field_name] = forms.CharField(
+                    label=field.name,
+                    required=False,
+                    widget=forms.Textarea(attrs={"rows": 3,}),
+                )
+                if field.is_mandatory:
+                    self.fields[field_name].required = True
+            self.fields["amend_custom_properties"].required = False
+
+    def save(self, commit=True):
+        permit_request = super().save(commit=False)
+
+        prefix = models.AMEND_CUSTOM_FIELDS_PREFIX
+        amend_props = {}
+        for key, value in self.cleaned_data.items():
+            if key.startswith(prefix) and value:
```

Comment:
En faisant comme ça tu as le risque qu’un champ de ton formulaire qui n’est pas un champ custom ait ce préfixe. Ce serait plus propre d’avoir une méthode genre `get_custom_amend_field_name()` qui renvoie le nom du champ pour un champ custom. Par exemple:

```python
@staticmethod
def get_custom_amend_field_name(field):
    return f"amend_custom_field_{field.id}"

def save(self, commit=True):
    custom_amend_fields = services.get_permit_request_amend_custom_properties(self.instance)
    
    for field, field_name in [(field, self.get_custom_amend_field_name(field)) for field in custom_amend_fields]:
        amend_props[field.id] = self.cleaned_data[field_name]
```

Et tu pourrais aussi utiliser `get_custom_amend_field_name` dans le `__init__`, et enlever du coup `models.AMEND_CUSTOM_FIELDS_PREFIX` et simplement appeler `PermitRequestAdditionalInformationForm.get_custom_amend_field_name()`.

Reply:
Not needed! Refactored